src/Radicle/Internal/Eval.hs

```diff
@@ -62,7 +62,10 @@ specialForms = Map.fromList $ first Ident <$>
               Vec atoms_ -> do
                 atoms <- traverse isAtom (toList atoms_) ?? toLangError (SpecialForm "fn" "One of the arguments was not an atom")
                 e <- gets bindingsEnv
-                pure (Lambda atoms (b :| bs) e)
+                pure (Lambda (PosArgs atoms) (b :| bs) e)
+              Atom name -> do
+                e <- gets bindingsEnv
```

Comment:
The `e <- gets bindingsEnv` can be moved above the case analysis on `args`.

Reply:
👍🏻updated